test/branch_test.rb

```diff
@@ -116,8 +116,8 @@ def test_create_unicode_branch_nfc
     new_branch = @repo.create_branch(branch_name, "5b5b025afb0b4c913b4c338a42934a3863bf3644")
     refute_nil new_branch
 
-    assert_equal branch_name, new_branch.name
-    assert_equal "refs/heads/#{branch_name}", new_branch.canonical_name
+    assert_equal branch_name.force_encoding("ascii-8bit"), new_branch.name
+    assert_equal "refs/heads/#{branch_name}".force_encoding("ascii-8bit"), new_branch.canonical_name
```

Comment:
Can you use `.b` rather than `force_encoding`?  I'm not sure if that exists on all Ruby versions we support (but it might).